src/wallet/wallet2.cpp

```diff
@@ -1202,12 +1202,27 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
   std::list<cryptonote::block_complete_entry> blocks;
   std::vector<COMMAND_RPC_GET_BLOCKS_FAST::block_output_indices> o_indices;
 
+  std::string daemon_height_err = "";
+  uint64_t daemon_bc_height = this->get_daemon_blockchain_height(daemon_height_err);
+  if(daemon_height_err.size() > 0) {
+    throw std::runtime_error(daemon_height_err);
+  }
+
   // pull the first set of blocks
   get_short_chain_history(short_chain_history);
   m_run.store(true, std::memory_order_relaxed);
   if (start_height > m_blockchain.size() || m_refresh_from_block_height > m_blockchain.size()) {
-    if (!start_height)
-      start_height = m_refresh_from_block_height;
+    
+    std::string daemon_target_err = "";
+    uint64_t daemon_target_height = this->get_daemon_longest_known_chain_height(daemon_target_err);
+    if(daemon_target_err.size() > 0) {
+      throw std::runtime_error(daemon_target_err);
+    }    
+    
+    if (m_refresh_from_block_height > daemon_target_height) m_refresh_from_block_height = daemon_target_height;
```

Comment:
That seems wrong. daemon_target_height would be a blockchain height, so you'd likely need m_refresh_from_block_height = daemon_target_height - 1;